Automattic/wp-calypso
# Jetpack pricing: make tabs accessible

### Changes proposed in this Pull Request

In the [Jetpack pricing page](https://cloud.jetpack.com/pricing),  the Products/Bundles toggle is implemented as a radio group, while it is effectively a tab component. This PR fixes it, so that keyboard interaction matches the expected tab behaviour (see videos below), and so that the [Info & Relationships accessibility guideline](https://www.w3.org/TR/WCAG21/#info-and-relationships) is respected.

### Implementation notes

This PR does several things:
- Replace `ViewFilter` by the [`TabPanel` component](https://github.com/Automattic/gutenberg/tree/trunk/packages/components/src/tab-panel), that implements most of the accessibility requirements.
- Remove the `ViewFilter` component, that's not used anymore.
- Remove the `ViewFilterProps` type.
- Make the Products/Bundles tab at least 44px high, so that it matches the minimum clickable/touchable area requirements.

When a tab has focus, pressing the down arrow should give focus to its matching panel. I'll submit a later PR to update the component.

### Testing instructions

Please test in both WordPress.com and Jetpack Cloud.

- Spin up the pricing page, by running this branch locally, or with one of the live links below.
- Visit the pricing page.
- Check that the Products/Bundles tab looks good and works as expected on both desktop and mobile.
- Check that the products list is rendered by default, and when the `view` query parameter is set to `products`.
- Check that the bundles list is rendered when the `view` query parameter is set to `bundles`.
- Check that when switching tab, the `view` parameter in the URL bar is updated accordingly.
- Use the keyboard to navigate the page. Once the first tab has focus, pressing `Tab` should give focus to the first element in the tab panel, while pressing the left or right arrow should give focus to the previous/next tab (see video belows).

### Screenshots

_Before_
<img width="377" alt="Screen Shot 2022-10-06 at 1 26 37 PM" src="https://user-images.githubusercontent.com/1620183/194378982-08ea8c9d-ce27-44c1-80b4-8a3a271dce9d.png">

https://user-images.githubusercontent.com/1620183/194382990-a4ef7040-de02-49c5-b991-899a45e83158.mov

_When the Products button has focus, pressing `Tab` moves the focus to the Bundles button._


_After_
<img width="370" alt="Screen Shot 2022-10-06 at 1 30 45 PM" src="https://user-images.githubusercontent.com/1620183/194379696-6e8a4690-54dd-4d8b-a608-2e7c38f4c994.png">


https://user-images.githubusercontent.com/1620183/194383232-545c1dd7-36c1-4cf5-9a85-53faec6f0921.mov

_When the Products tab has focus, pressing `Tab` moves the focus to the first element of the associated Panel. Pressing the right arrow moves the focus to the next tab. That's the expected behaviour for tab navigation._
